docs: add missing docs for array rule (#28437)
ant-design/ant-design: components/form/index.en-US.md

| validateTrigger | Set validate trigger event. Must be the sub set of `validateTrigger` in Form.Item | string \| string\[] |
| validator | Customize validation rule. Accept Promise as return. See [example](#components-form-demo-register) | ([rule](#Rule), value) => Promise |
| whitespace | Failed if only has whitespace | boolean |

> | defaultField | Only valid when `type` is `array`, used to validate the rules of array elements | [rule](#Rule) |

Reviewer: ```suggestion
| defaultField | Validate rule for all array elements, valid when `type` is `array` | [rule](#Rule) |
```

Reviewer: 还有个 fields 可以写一下：https://github.com/yiminghe/async-validator/#deep-rules
